tests/60app-services/02ghost.pl

```diff
@@ -87,6 +87,72 @@
       })->then_done(1);
    };
 
+test "Application services can be not rate limited",
+   requires => [ as_ghost_fixture(), $main::AS_USER,
+                     room_fixture( requires_users => [ $user_fixture ] ),
+                qw( can_receive_room_message_locally )],
+
+   do => sub {
+      my ( $ghost, $as_user, $room_id ) = @_;
+
+      Future->needs_all(
+         ( map {
+            await_as_event( "m.room.member" )->then( sub { Future->done( 1 ); } ),
+         } ( 0, 1 )),
+
+         do_request_json_for( $as_user,
+            method => "POST",
+            uri    => "/api/v1/rooms/$room_id/join",
+            params => {
+               user_id => $ghost->user_id,
+            },
+
+            content => {},
+         ),
+
+         do_request_json_for( $as_user,
+            method => "POST",
+            uri    => "/api/v1/rooms/$room_id/join",
+            params => {
+               user_id => $as_user->user_id,
+            },
+
+            content => {},
+         ),
+      )->then( sub {
+         Future->needs_all(
+            ( map {
+               await_as_event( "m.room.message" )->then( sub { Future->done( 1 ); } )
+            } 0 .. 300 ),
```

Comment:
There's an off-by-one in this code. The range `0 .. 300` is inclusive at both ends, so has 301 elements. Similarly in the message sending you send 151 messages from each user; a total of 302.

I'd suggest using the ranges `1 .. 300` and `1 .. 150` here